src/crypto/EncryptionSetup.ts

```diff
@@ -365,7 +365,7 @@ class SSSSCryptoCallbacks {
         }
         // if we don't have the key cached yet, ask
         // for it to the general crypto callbacks and cache it
-        if (this.delegateCryptoCallbacks) {
+        if (this.delegateCryptoCallbacks?.getSecretStorageKey) {
```

Comment:
According to the constructor, we shouldn't need this `?` as `delegateCryptoCallbacks` must be a `ICryptoCallbacks`.

There's another issue that
* either the constructor's signature of `SSSSCryptoCallbacks` and `EncryptionSetupBuilder` should allow `ICryptoCallbacks|undefined` or
* [`MatrixClient.cryptoCallbacks`](https://github.com/matrix-org/matrix-js-sdk/blob/a69d9a9f20b680e3710e0f43cca68a2ce31ba030/src/client.ts#L315) must not be undefined.

It's handed over here.
https://github.com/matrix-org/matrix-js-sdk/blob/a69d9a9f20b680e3710e0f43cca68a2ce31ba030/src/crypto/index.ts#L616-L620

Reply:
https://github.com/matrix-org/matrix-js-sdk/pull/1847/files#diff-2ed9f095536152a0f96b01826f2bc6fa8d04591b7c5bc02782804d4e12bdbc31R382

regards `this.delegateCryptoCallbacks` as optional, so I guess version 1 is/was supposed to be the truth.

Reply:
As that's yet a different issue, I solved that in a separate PR: #1849

Reply:
@jaller94 so let's go back to checking `this?.delegateCryptoCallbacks?.getSecretStorageKey`. You can just click commit suggestion. 😄

Reply:
I don't agree. According to our types, `this` and `this.delegateCryptoCallbacks` are safe.
However, it doesn't hurt to add the question marks.

```suggestion
        if (this?.delegateCryptoCallbacks?.getSecretStorageKey) {
```